Docs/checklist-for-new-user-facing-features.md

```diff
@@ -0,0 +1,47 @@
+## New Feature Development Checklist
+
+So you've added a new feature. Users will see it!
+
+You've landed the code, it's indented properly, it looks lovely! What's Next!?
+
+###  Has your feature got tests yet?
+
+  * perhaps existing tests will cover it, may be adding another unit test will be enough.
+  * If you're adding new UI:
+    - ensure widgets have `accessbilityIdentifiers`. 
+    - Add it to FxScreenGraph.swift.
+    - XCUITests should be used, but as a last resort.
+
+### Should your new feature be keyboard navigable?
+
+  * Adding UIKeyCommands will help.
+  * Getting to the new features is as important as using your feature.
+
+### Accessibility for new UI
+
+  * Add accessibilityLabels
+  * Have you tested it with VoiceOver?
+
+### Has your feature got new Strings?
+
+  * Is it `NSLocalizable`? 
+  * It should probably go in `Strings.swift`. Ask about String export deadlines.
+  * Are the strings screenshotted? The `L10nScreenshotTests` are there to help diagnose layout issues caused by missing or long strings and right to left issues.
+
+### How do you know if your new feature is used?
+
+  * Add event telemetry
+  * Consider performance telemetry.
+  * Add a Telemetry dashboard — this will help make decisions about your feature in the future.
+
+### You're feature is awesome! Have you told anyone outside of Engineering or UX about it?
+
+  * Should it go on the What's New page? (to tell the users)
+  * Should it be documented?
+  * Have you told These Weeks in Mobile Firefox?
+  * Have you told Marketing?
```

Comment:
AFAIK, communicating the above items is ad-hoc, would be nice if said exactly how we would like this done.

Reply:
Do we need a rolling a `release_notes.txt` in our project? So that we do not have to scramble the week before a release to get a What's New (app store or mozilla.org) together?